tests/unit/generate_detect_target_contour.py

```diff
@@ -0,0 +1,183 @@
+            blur: Indicates whether the landing pad should have a blur effect. default: False.
+            angle (int): The rotation angle of the landing pad in degrees. defaults: 0.
+        """
+        self.center = center
+        self.axis = axis
+        self.blur = blur
+        self.angle = angle
+
+
+class NumpyImage:
+    def __init__(self, image: np.ndarray):
+        """
+        Holds the Numpy Array which represents an image.
+        """
+        self.image = image
+
+
+class BoundingBox:
+    def __init__(self, top_left: tuple[int, int], bottom_right: tuple[int, int]):
+        """
+        Holds the data that define the generated bounding boxes.
+
+        Attributes:
+            top_left: A tuple of integers that represents top left corner of bounding box.
+            bottom_right: A tuple of integers that represents bottom right corner of bounding box.
+        """
+        self.top_left = top_left
+        self.bottom_right = bottom_right
+
+
+class LandingPadTestData:
+    def __init__(self, image: NumpyImage, boxes_list: np.ndarray):
+        """
+        Struct to hold the data needed to perform the tests.
+
+        Attributes:
+            image = A numpy array that represents the image needed to be tested
+            bounding_box_list: A numpy array that holds a list of expected bounding box coordinates
+        """
+        self.image = image.image
+        self.bounding_box_list = boxes_list
+
+
+def add_blurred_landing_pad(background: np.ndarray, landing_data: LandingPadData) -> NumpyImage:
+    """
+    Blurs an image a singular lading pad, adds it to the background.
+
+    Attributes:
+        image = A numpy array that represents background.
+        landing_data = The landing pad which is to be blurred.
+    Returns:
+        NumpyImage: A numpy array of the new blured image.
+    """
+    x, y = background.shape[:2]
+
+    mask = np.zeros((x, y), np.uint8)
+    mask = cv2.ellipse(
+        mask,
+        landing_data.center,
+        landing_data.axis,
+        landing_data.angle,
+        0,
+        360,
+        255,
+        -1,
+    )
+
+    mask = cv2.blur(mask, (25, 25), 7)
+
+    alpha = mask[:, :, np.newaxis] / 255.0
+    # Brings the image back to its original color
+    fg = np.full(background.shape, LANDING_PAD_COLOR_BLUE, dtype=np.uint8)
+
+    blended = (background * (1 - alpha) + fg * alpha).astype(np.uint8)
+    return NumpyImage(blended)
+
+
+def draw_landing_pad(
+    image: np.ndarray, landing_data: LandingPadData
+) -> tuple[NumpyImage, BoundingBox]:
+    print("asdasd")
+    print(image)
+    """
+    Draws an singular landing pad on the provided image and saves the bounding box coordinates to a text file.
+
+    Attributes:
+        image = A numpy array that represents background
+        landing_data = The landing pad which is to be placed
+    Returns:
+        NumpyImage: A numpy array of the new image .
+        BoundingBox: Bounding box of the newly placed bounding box.
```

Comment:
`Return: Image with landing pad and the bounding box for the drawn landing pad.`